test-framework/sudo-compliance-tests/src/sudo/sudoers/specific_defaults.rs

```diff
@@ -0,0 +1,256 @@
+        .as_user(USERNAME)
+        .output(&env)?;
+    assert!(!output.status().success());
+
+    // Root password is accepted for any other runas
+    let output = Command::new("sh")
+        .arg("-c")
+        .arg(format!("echo {ROOT_PASSWORD} | sudo -S true"))
+        .as_user(USERNAME)
+        .output(&env)?;
+    assert!(output.status().success());
+
+    Ok(())
+}
+
+#[test]
+fn rootpw_can_be_per_general_command() -> Result<()> {
+    const PASSWORD: &str = "passw0rd";
+    const ROOT_PASSWORD: &str = "r00t";
+
+    let env = Env(format!(
+        "Defaults!/usr/bin/tr* rootpw
+        Defaults passwd_tries=1
+        {USERNAME} ALL=(ALL:ALL) ALL"
+    ))
+    .user_password("root", ROOT_PASSWORD)
+    .user(User(USERNAME).password(PASSWORD))
+    .build()?;
+
+    // Root password is not accepted for 'whoami'
+    let output = Command::new("sh")
+        .arg("-c")
+        .arg(format!("echo {ROOT_PASSWORD} | sudo -S whoami"))
+        .as_user(USERNAME)
+        .output(&env)?;
+    assert!(!output.status().success());
+
+    // Root password is accepted for 'true'
+    let output = Command::new("sh")
+        .arg("-c")
+        .arg(format!("echo {ROOT_PASSWORD} | sudo -S true args"))
+        .as_user(USERNAME)
+        .output(&env)?;
+    assert!(output.status().success());
+
+    Ok(())
+}
+
+#[test]
+fn rootpw_can_be_per_command_w_args() -> Result<()> {
+    const PASSWORD: &str = "passw0rd";
+    const ROOT_PASSWORD: &str = "r00t";
+
+    let env = Env(format!(
+        "Cmnd_Alias TRUE=/usr/bin/true ignored
+         Defaults!TRUE rootpw
+         Defaults passwd_tries=1
+         {USERNAME} ALL=(ALL:ALL) ALL"
+    ))
+    .user_password("root", ROOT_PASSWORD)
+    .user(User(USERNAME).password(PASSWORD))
+    .build()?;
+
+    // Root password is not accepted for 'whoami'
+    let output = Command::new("sh")
+        .arg("-c")
+        .arg(format!("echo {ROOT_PASSWORD} | sudo -S true bla"))
+        .as_user(USERNAME)
+        .output(&env)?;
+    assert!(!output.status().success());
+
+    // Root password is accepted for 'true'
+    let output = Command::new("sh")
+        .arg("-c")
+        .arg(format!("echo {ROOT_PASSWORD} | sudo -S true ignored"))
+        .as_user(USERNAME)
+        .output(&env)?;
+    assert!(output.status().success());
+
+    Ok(())
+}
+
+//note: we don't repeat all of the above combinations, the following tests
+//focus on the peculiar behaviour of 'secure_path'
+
+#[test]
+fn securepath_can_be_per_user() -> Result<()> {
+    const PASSWORD: &str = "passw0rd";
+
+    let env = Env(format!(
+        "Defaults secure_path=\"/usr/bin\"
+        Defaults:{USERNAME} secure_path=\"/user/\"
+        ALL ALL=NOPASSWD: ALL"
+    ))
+    .user(User(USERNAME).password(PASSWORD))
+    .build()?;
+
+    // Command is not found (/root/true does not exist)
+    let output = Command::new("sh")
+        .arg("-c")
+        .arg("sudo true")
```

Comment:
Why is this using `sh -c 'sudo true'` rather than `sudo true`?

Reply:
Because it's copied from other tests. I'll fix this ;-)